test/com/google/enterprise/adaptor/database/ResponseGeneratorTest.java

```diff
@@ -181,13 +255,50 @@ public void testBlobColumnModeIncorrectColumnName() throws Exception {
         new Class[] { Response.class }, bar);
     Map<String, String> cfg = new TreeMap<String, String>();
     cfg.put("columnName", "my-col-name-is-wrong");
-    ResponseGenerator resgen = ResponseGenerator.blobColumn(cfg);
+    ResponseGenerator resgen = ResponseGenerator.contentColumn(cfg);
     String content = "hello world";
+    ArrayList<String> names = new ArrayList<String>(Arrays.asList("my-col-name-is-wrong"));
+    ArrayList<Integer> types = new ArrayList<Integer>(Arrays.asList(Types.BLOB));
     byte b[] = content.getBytes("US-ASCII");
     thrown.expect(java.sql.SQLException.class);
-    resgen.generateResponse(makeMockBlobResultSet(b), response);
+    resgen.generateResponse(makeMockBlobResultSet(b, names, types), response);
   }
- 
+
+  @Test
+  public void testClobColumnModeServesResultClob() throws Exception {
+    MockResponse bar = new MockResponse();
+    Response response = (Response) Proxy.newProxyInstance(
+        Response.class.getClassLoader(),
+        new Class[] { Response.class }, bar);
+    Map<String, String> cfg = new TreeMap<String, String>();
+    cfg.put("columnName", "my-clob-col");
+    ResponseGenerator resgen = ResponseGenerator.contentColumn(cfg);
+    String content = "hello world";
+    char c[] = content.toCharArray();
```

Comment:
What benefit is the char array playing here over passing the string to makeMockClobResultSet? If it's just for SerialClob's benefit, don't pollute the callers with that implementation detail.